### ISSUE TITLE
Carriage problems

### ISSUE THREAD

**Issue by davidwengier** [davidwengier/trains2#1]
* FIx isTrainHere so that it accounts for gaps in tracks between carriages.
* Add getBackCoords() method that is carriage aware, and allows a buffer at the end for collisions
* Carriages should get their speed from the train, to ensure its consistent
* Carriages should probably also always get their position based off the train, so its impossible for gaps to open up.

**Comment by ScottHolden** [comment]
We could also then even represent the carriage drawing routine as 2 points on a track (one for each wheel pivot point), instead of a single point with an angle. This would better mimic how a train goes round a corner in the real world :)

**Comment by davidwengier** [comment]
Yeah, having two points makes sense for hit testing too as per #15. There wouldn't be any gaps between carriages if we properly knew where they were

**Comment by ScottHolden** [comment]
Started work within #22
